src/devices/Bmxx80/Bme680.cs

```diff
@@ -0,0 +1,223 @@
+        public Task<double> ReadPressureAsync()
+        {
+            // Read pressure data.
+            byte lsb = Read8BitsFromRegister((byte)Bme680Register.PRESSUREDATA_LSB);
+            byte msb = Read8BitsFromRegister((byte)Bme680Register.PRESSUREDATA_MSB);
+            byte xlsb = Read8BitsFromRegister((byte)Bme680Register.PRESSUREDATA_XLSB);
+
+            // Convert to a 32bit integer.
+            var adcPressure = (msb << 12) + (lsb << 4) + (xlsb >> 4);
+
+            return Task.FromResult(CompensatePressure(adcPressure));
+        }
+
+        /// <summary>
+        /// Read the temperature.
+        /// </summary>
+        /// <returns>Calculated temperature.</returns>
+        public Task<Temperature> ReadTemperatureAsync()
+        {
+            // Read temperature data.
+            byte lsb = Read8BitsFromRegister((byte)Bme680Register.TEMPDATA_LSB);
+            byte msb = Read8BitsFromRegister((byte)Bme680Register.TEMPDATA_MSB);
+            byte xlsb = Read8BitsFromRegister((byte)Bme680Register.TEMPDATA_XLSB);
+
+            // Convert to a 32bit integer.
+            var adcTemperature = (msb << 12) + (lsb << 4) + (xlsb >> 4);
+
+            return Task.FromResult(CompensateTemperature(adcTemperature));
+        }
+
+        /// <summary>
+        /// Compensates the humidity.
+        /// </summary>
+        /// <param name="temperature">The temperature to use.</param>
+        /// <param name="adcHumidity">The humidity value read from the device.</param>
+        /// <returns>The percentage relative humidity.</returns>
+        private double CompensateHumidity(double temperature, int adcHumidity)
+        {
+            // Calculate the humidity.
+            var var1 = adcHumidity - ((_bme680Calibration.DigH1 * 16.0) + ((_bme680Calibration.DigH3 / 2.0) * temperature));
+            var var2 = var1 * ((_bme680Calibration.DigH2 / 262144.0) * (1.0 + ((_bme680Calibration.DigH4 / 16384.0) * temperature)
+                + ((_bme680Calibration.DigH5 / 1048576.0) * temperature * temperature)));
+            var var3 = _bme680Calibration.DigH6 / 16384.0;
+            var var4 = _bme680Calibration.DigH7 / 2097152.0;
+            var calculatedHumidity = var2 + ((var3 + (var4 * temperature)) * var2 * var2);
+
+            if (calculatedHumidity > 100.0)
+            {
+                calculatedHumidity = 100.0;
+            }
+            else if (calculatedHumidity < 0.0)
+            {
+                calculatedHumidity = 0.0;
+            }
+
+            return calculatedHumidity;
+        }
+
+        /// <summary>
+        /// Compensates the pressure.
+        /// </summary>
+        /// <param name="adcPressure">The pressure value read from the device.</param>
+        /// <returns>The pressure in Pa.</returns>
+        private double CompensatePressure(int adcPressure)
+        {
+            // Calculate the pressure.
+            var var1 = (TemperatureFine / 2.0) - 64000.0;
```

Comment:
Will TemperatureFine have the right value here? Shouldn't CompensatePressure call CompensateTemperature first in order to ensure that TemperatureFine has been correctly initialized?

Reply:
Ah, yes it should!